constellation-app/constellation
# Issue 57 allow pinning of nodes to the graph

### Description of the Change

- Added a position locked attribute to vertexes.
- Added actions/pllugins to allow selected vertexes to have their position_locked attributes set to true or false.
- Modified AbstractInclusionGraph so that when determining selected vertexes (in core arrangements) any vertex marked as position_locked is ignored (hence will NOT be moved).

### Alternate Designs

nil

### Why Should This Be In Core?

This was a requested feature by existing user base.

### Benefits

Makes it easier for users to control the arrangement of displays.

### Possible Drawbacks

If user is NOT aware a vertex is locked, they may need to investigate why it doesnt get arranged.

### Verification Process

1. Create a graph
2. Add a heap of nodes
3. Use the lock toolbar option to lock several nodes
4. Perform some arrangements and ensure the locked vertexes don't move (test with a subset of selected vertexes, no vertexes selected, and all vertexes selected) 
5.  Repeat step 4 using locked menu item
6. Use toolbar and menu to unlock some vertexes, and re-arrange, confirming only unlocked vertexes move.

### Applicable Issues

#57
